### ISSUE TITLE
Ambiant voices issues with Amarant/Freya battlecommand

### ISSUE THREAD

**Issue by Algarad** [Albeoris/Memoria#386]
I have tried different combination, but I cannot find the right condition for:

- Amarant Throw command
- Amarant Flair command
- Freya Jump command

https://github.com/Albeoris/Memoria/blob/main/Assembly-CSharp/Memoria/Data/Battle/BattleCommandId.cs

I have no idea for the Flair command, while there are self-explanatory entries for Jump or Launch in BattleCommandId that do not work though, or I cannot get them to work.

Thank you

**Comment by Algarad** [comment]
Thank you for the hints,
my bad for "Throw" - Stupid misspell by my side.

I have tried but `WhenCommandPerform` and `WhenHitEffect` as you suggested, but both seems not working:
`>Act Freya
WhenCommandPerform
VoicePath:Battle/Freya/Jump1,Jump2,Jump3
[Condition] CommandId == BattleCommandId_Jump || CommandId == BattleCommandId_Jump2 [/Condition]

>Act Freya
WhenHitEffect
VoicePath:Battle/Freya/JumpAttack1,JumpAttack2,JumpAttack3
[Condition] CommandId == BattleCommandId_JumpAttack || CommandId == BattleCommandId_JumpTrance [/Condition]`

`BattleCommandId_Jump` works fine with `WhenCommandInput` (of course, it just triggers immediately).

**Comment by Tirlititi** [comment]
I'll make sure that these commands (Jump / Spear) are handled correctly by the VA system. I don't see why it doesn't trigger when looking at the code but there has to be a reason.

**Comment by Tirlititi** [comment]
So, the problem was that Freya is prevented to have any voice acting when jumping (which is a special status ailment). There are multiple statuses like that. For example, Silence prevents any voice acting except those related to Silence (ie. when the status is inflicted, when it's cured and when Silence breaks a spell command issued before getting the status).

I wanted to keep this behaviour. I think it's better to have the Jump status prevent voice acting as the rule and exceptions must be explicit (eg. so Freya doesn't have a victory speech if she's in the air).

Thus, for technical reasons, it's not ``>Act`` that must be used for the Jump / Spear commands but rather ``>StatusChange``:
```
>StatusChange Freya
VoicePath:...
Status:Jump
WhenAdded

>StatusChange Freya
VoicePath:...
Status:Jump
WhenUsed
```
The first one ``WhenAdded`` is at the time of the jump, while the second one ``WhenUsed`` is at the time of the Spear attack. Both are played early on (at the start of the jump / spear movement). You can add conditions to check if Freya is in trance or things like that like for any other battle voice effect.

Hopefully that's good.
Maybe the triggering conditions when Freya is in Jump-Trance could be improved: as it is now, ``WhenAdded`` triggers again after the multi-spear hit if Freya stays in the air.

**Comment by Algarad** [comment]
Hi,
I have tested and can confirm that `WhenAdded` with the status change works perfectly. 
`WhenUsed` is not working though, however `WhenRemoved` works (Freya says her words when she return to her position).

`>StatusChange Freya
Status:Jump
WhenAdded
VoicePath:Battle/Freya/Jump1,Jump2,Jump3

>StatusChange Freya
Status:Jump
WhenRemoved
VoicePath:Battle/Freya/JumpAttack1,JumpAttack2,JumpAttack3`

Thank you for you clarification!

**Comment by Tirlititi** [comment]
Yes, you need to wait the next release for ``WhenUsed``, as I still had to fix a problem about that in the commit [2c7071c](https://github.com/Albeoris/Memoria/commit/2c7071c045661c686eb6d05a84ff221941fa5950)
